cache.go

```diff
@@ -818,6 +818,34 @@ func (c *cache) delete(k string) {
 	delete(c.items, k)
 }
 
+func (c *cache) Expire(k string, d time.Duration) error {
```

Comment:
To avoid code duplication I suggest to replace the method implementation with the following code:

```go
func (c *cache) Expire(k string, d time.Duration) error {
	e := time.Now().Add(d)
	return c.ExpireAt(k, e)
}
```

Also, I would rename this method to 'ExpireAfter'.